contract/src/interfaces/IChannel.sol

```diff
@@ -8,19 +8,28 @@ import {Channel, State} from "./Types.sol";
  * @notice Main interface for the state channel system
  */
 interface IChannel {
-    event ChannelPartiallyFunded(bytes32 indexed channelId, Channel channel);
-    event ChannelOpened(bytes32 indexed channelId, Channel channel);
-    event ChannelChallenged(bytes32 indexed channelId, uint256 expiration);
-    event ChannelCheckpointed(bytes32 indexed channelId);
-    event ChannelClosed(bytes32 indexed channelId);
+    event Created(bytes32 indexed channelId, Channel channel, Amount[] expected);
```

Comment:
Why do we need an `Amount[]` only for creator? I would rather we remove it alltogether, replaced with `Allocations[]` or add to `Joined` too. Although it is a bad practice in Solidity in general to put Structs in events

Reply:
Here we need to return the whole state, so other participants would know, what they need to sign

Reply:
Return the whole state, its a mistake of old code

Reply:
1fcd517fd0752ce0986130383b1cda29faedf515